lib/omniauth/strategies/indieauth.rb

```diff
@@ -2,76 +2,131 @@
 
 require 'omniauth'
 require 'faraday'
+require 'faraday_middleware'
+require 'indieauth_discovery/profile'
 require 'cgi'
 
+require 'omniauth/indieauth/errors'
+
 module OmniAuth
   module Strategies
     # IndieAuth strategy for OmniAuth.
     class IndieAuth
       include OmniAuth::Strategy
 
-      option :name, 'indieauth'
-      option :server, 'https://indieauth.com'
+      option :name, :indieauth
       option :client_id
 
-      attr_accessor :me
-
-      def redirect_uri
-        full_host + script_name + callback_path
-      end
+      attr_reader :profile
 
       def request_phase
-        puts redirect_uri
-        redirect "#{options.server}/sign-in?redirect_uri=#{URI.encode_www_form_component(redirect_uri)}&client_id=#{URI.encode_www_form_component(options.client_id)}"
+        valid_request_phase_params? ? authorize : profile_url_form
```

Comment:
This is personal preference, but I tend to use guard clauses instead of ternaries:

```rb
return authorize if valid_phase_params?

profile_url_form
```

Reply:
I actually prefer guard clauses typically instead of ternaries as well, not sure why I flip-flopped here :)